src/Text/recipes.jl

```diff
@@ -0,0 +1,44 @@
+"""
+    TextFolders(textfile; labelfn = parentname, split = false)
+
+Recipe for loading a single-label text classification dataset
+stored in hierarchical folder format. 
+"""
+Base.@kwdef struct TextFolders <: Datasets.DatasetRecipe
+    labelfn = parentname
+    split::Bool = false
+    filefilterfn = _ -> true
+end
+
+Datasets.recipeblocks(::Type{TextFolders}) = Tuple{TextBlock, Label}
+
+function Datasets.loadrecipe(recipe::TextFolders, path)
+    isdir(path) || error("$path is not a directory")
+    data = loadfolderdata(
+        path,
+        filterfn=f -> istextfile(f) && recipe.filefilterfn(f),
+        loadfn=(loadfile, recipe.labelfn),
+        splitfn=recipe.split ? grandparentname : nothing)
+
+    (recipe.split ? length(data) > 0 : nobs(data) > 0) || error("No text files found in $path")
+
+    labels = recipe.split ? first(values(data))[2] : data[2]
+    blocks = TextBlock(), Label(unique(eachobs(labels)))
```

Comment:
Was this meant to be 
```suggestion
    blocks = (TextBlock(), Label(unique(eachobs(labels))))
```
? Given the lines below, I'd recommend declaring them separately and doing the tuple packing in the return.

Reply:
Even this is how I've seen it was implemented in `Vision/recipes.jl`. But if it seems reasonable to do this way, we can change it.